pkg/queue/timeout.go

```diff
@@ -74,29 +74,26 @@ func (h *timeoutHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	tw := &timeoutWriter{w: w}
 	go func() {
+		// The defer statements are executed in LIFO order,
+		// so recover will execute first, then only, the channel will be closed.
+		defer close(done)
 		defer func() {
 			if p := recover(); p != nil {
 				panicChan <- p
 			}
 		}()
 		h.handler.ServeHTTP(tw, r.WithContext(ctx))
-
-		// Closing the channel is not deferred to give the panic recovery
-		// precedence and not successfully complete the request by accident.
-		close(done)
 	}()
 
 	timeout := time.After(h.dt)
 	for {
 		select {
 		case p := <-panicChan:
-			close(done)
 			panic(p)
 		case <-done:
 			return
 		case <-timeout:
 			if tw.TimeoutAndWriteError(h.errorBody()) {
-				cancelCtx()
```

Comment:
This feels like a semantic change.

Reply:
There's a `defer cancelCtx()` at the top of this method

https://github.com/knative/serving/pull/3244/files#diff-5d9c31568bce8bea685733b9a8423925L67

Reply:
Yes, I thinnk this was a double invoke.

Reply:
gotcha, that looks right.